packages/x-date-pickers/src/AdapterMomentJalaali/AdapterMomentJalaali.ts

```diff
@@ -186,6 +190,18 @@ export class AdapterMomentJalaali extends AdapterMoment implements MuiPickersAda
     return this.moment(value).isSame(comparing);
   };
 
+  public isSameYear = (value: Moment, comparing: Moment) => {
```

Comment:
This was using the `month` and `year` instead of `jMonth` and `jYear` which was just wrong